src/spoke/Spoke.sol

```diff
@@ -973,9 +993,9 @@ abstract contract Spoke is ISpoke, Multicall, NoncesKeyed, AccessManagedUpgradea
   function _getUserDebt(
     IHubBase hub,
     uint256 assetId,
-    UserPosition storage userPosition
+    UserPosition storage userPosition,
+    uint256 drawnIndex
```

Comment:
might be worth to have 2 internal method so we don't need to fetch and pass `drawnIndex` when we already pass the `hub` + `assetId`, or remove `hub` & `assetId` params here as it seems we do not need them anymore.